external/src/main/java/com/redhat/cloud/policies/engine/process/LightweightEngine.java

```diff
@@ -0,0 +1,206 @@
+            policiesAction.setTimestamp(nowUTC);
+
+            PoliciesAction.Context context = policiesAction.getContext();
+            context.setSystemCheckIn(LocalDateTime.from(DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(event.getContext().get("check_in"))));
+            context.setInventoryId(event.getContext().get("inventory_id"));
+            context.setDisplayName(event.getTags().get("display_name").iterator().next());
+
+            for (Policy policy : accountPolicies) {
+                if (policy.enabled) {
+                    alertsCounter.inc();
+
+                    LOGGER.debugf("Starting the policy condition evaluation");
+                    if (policy.condition == null || policy.condition.trim().isEmpty() || ExprParser.evaluate(event, policy.condition)) {
+                        policiesHistoryRepository.create(policy.id, event);
+                        if (shouldSendNotification(policy)) {
+
+                            // Tags from all events are merged into the policies action context tags.
+                            for (Map.Entry<String, String> tagEntry : event.getTags().entries()) {
+                                String tagValue = tagEntry.getValue();
+                                if (tagValue == null) {
+                                    // Same behavior as previously with JsonObjectNoNullSerializer with old hooks
+                                    tagValue = "";
+                                }
+                                context.getTags().computeIfAbsent(tagEntry.getKey(), unused -> new HashSet<>()).add(tagValue);
+                            }
+
+                            // Each event is added to the policies action as an event.
+                            PoliciesAction.Event policiesEvent = new PoliciesAction.Event();
+                            policiesEvent.getPayload().setPolicyCondition(policy.condition);
+                            policiesEvent.getPayload().setPolicyId(policy.id.toString());
+                            policiesEvent.getPayload().setPolicyName(policy.name);
+                            policiesEvent.getPayload().setPolicyDescription(policy.description);
+                            policiesAction.getEvents().add(policiesEvent);
+                        }
+                    }
+                }
+            }
+
+            if (!policiesAction.getEvents().isEmpty()) {
+                try {
+                    String payload = serializeAction(policiesAction);
+                    LOGGER.debugf("Sending Kafka payload %s", payload);
+                    emitter.send(buildMessageWithId(payload));
+                    notificationsCounter.inc();
+                } catch (IOException e) {
+                    LOGGER.warnf("Failed to serialize action for accountId", policiesAction.getAccountId());
+                }
+            }
+        }
+    }
+
+    private boolean shouldSendNotification(Policy policy) {
```

Comment:
maybe just "sendNotification"? I am using should very often in tests so it's clear what to expect from a test. 
but I am also fine with the current name. Do it as you like :)

Reply:
I don't know, `sendNotification` sounds like the method will actually do that while it won't.

Reply:
true. Naming is hard.. I also don't know a better name. so leave it as it is.